test/tests/misc/check_error/tests

```diff
@@ -886,6 +886,14 @@
     requirement = 'The system shall report an error when an attempt is made to couple a variable with itself.'
     issues = '#10398'
   [../]
+  [coupling_itself_ad]
+    type = 'RunException'
+    input = 'coupling_itself.i'
+    cli_args = 'Kernels/coupled/type=ADCoupledForce'
+    expect_err = "Coupled variable 'v' needs to be different from 'variable' with ADCoupledForce"
+    requirement = 'The system shall report an error when an attempt is made to couple a variable with itself in coupled force term using automated differentiation.'
+    issues = '#18214'
+  []
```

Comment:
Of course if we remove the restriction, then this test is not relevant. Up to you what you want to do with that. It might be nice to have a test that compares against a gold file that is sym-linked to a gold file generated with regular `CoupledForce`

Reply:
Another alternative test would be a PetscJacobianTest which would have shown a bad Jacobian due to use of `coupledValue` instead of `adCoupledValue`. I think that's probably more important than an `Exodiff` test

Reply:
Yeah, I know about a Jacobian test where I tried to use this Kernel and it failed. But, I thought it was because of combining FV and FE. I will tried that test again with these changes.